state/azure/cosmosdb/cosmosdb_query.go

```diff
@@ -0,0 +1,192 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package cosmosdb
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/a8m/documentdb"
+	"github.com/agrea/ptr"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/dapr/components-contrib/state"
+	"github.com/dapr/components-contrib/state/query"
+)
+
+type Query struct {
+	query documentdb.Query
+	limit int
+	token string
+}
+
+func (q *Query) VisitEQ(f *query.EQ) (string, error) {
```

Comment:
In addition to the query text, can this also contain the map of parameters as a response? I think we should support prepared statement natively, otherwise, user code would still be vulnerable to SQL injection if the value is hand crafted for that.

Reply:
The function argument `query.EQ` already contains the exact key/value. What is the purpose and content of the suggested map of parameters?